Source/WebGPU/WebGPU/BindGroupLayout.mm

```diff
@@ -141,13 +140,36 @@
     if (storageTexture.nextInChain)
         return nil;
 
-    UNUSED_PARAM(storageTexture);
     auto descriptor = [MTLArgumentDescriptor new];
     descriptor.dataType = MTLDataTypeTexture;
     // FIXME: Implement this.
     return descriptor;
 }
 
+bool BindGroupLayout::isPresent(const WGPUExternalTextureBindingLayout&)
+{
+    return true;
+}
+
+static MTLArgumentDescriptor *createArgumentDescriptor(const WGPUExternalTextureBindingLayout& externalTexture)
+{
+    if (externalTexture.nextInChain)
+        return nil;
+
+    // FIXME: Implement this properly.
+    // External textures have a bunch of information included in them, not just a single texture.
+    auto descriptor = [MTLArgumentDescriptor new];
+    descriptor.dataType = MTLDataTypeTexture;
+#if USE(METAL_ARGUMENT_ACCESS_ENUMS)
+ALLOW_DEPRECATED_DECLARATIONS_BEGIN
```

Comment:
`MTLBindingAccessReadOnly` has been in the SDK for some time now. Do we really still need the `ALLOW_DEPRECATED_DECLARATIONS_BEGIN / END` here if we were to change `USE_METAL_ARGUMENT_ACCESS_ENUMS` from:
```
#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED < 150000) \
    || (PLATFORM(IOS_FAMILY) && __IPHONE_OS_VERSION_MAX_ALLOWED < 180000) \
    || (PLATFORM(APPLETV) && __TV_OS_VERSION_MAX_ALLOWED < 180000) \
    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MAX_ALLOWED < 110000)
#define USE_METAL_ARGUMENT_ACCESS_ENUMS 1
#endif
```
to:
```
#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED < 140000) \
    || (PLATFORM(IOS_FAMILY) && __IPHONE_OS_VERSION_MAX_ALLOWED < 170000) \
    || (PLATFORM(APPLETV) && __TV_OS_VERSION_MAX_ALLOWED < 170000) \
    || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MAX_ALLOWED < 100000)
```

Reply:
Good idea. I'd like to do this in a follow-up patch, though.

Reply:
https://github.com/WebKit/WebKit/pull/14235